app/controllers/application_controller.rb

```diff
@@ -5,6 +5,138 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
+  get '/' do 
+    erb :index
+  end
+
+  get '/signup' do 
+    if session[:user_id]
+      redirect to 'tweets'
+    else
+      erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if params["username"].empty? || params["password"].empty? || params["email"].empty?
+      redirect to '/signup'
+    else
+      user = User.create(params)
+      session[:user_id] = user.id
+      redirect to '/tweets'
+    end
+  end
+
+  get '/login' do 
+    if session[:user_id]
+      redirect to '/tweets'
+    else
+      erb :'/users/login'
+    end
+  end
+
+  post '/login' do
+    user = User.find_by(username: params["username"])
```

Comment:
do you have a validation on the user model, so that when a user is created it has a unique username? Otherwise we might run into a problem here.